core/animated-entity.js

```diff
@@ -0,0 +1,58 @@
+import { create_float_buffer, create_index_buffer } from './context';
+import { Entity } from './entity';
+
+export class AnimatedEntity extends Entity {
+  constructor(options) {
+    super(options);
+
+    if (!this.frames || !this.frames.length) {
+      this.buffers = [];
+    }
+
+    this.current_frame = 0;
+    this.next_frame = 1;
+    this.current_tick = 0;
+
+    // I'll leave those in here as a foundation to our future documentation.
+    // this.frame_delay;
+    // this.frames
+
+  }
+
+  create_buffers() {
+    this.number_of_frames = this.frames.length;
+    this.buffers = this.frames.map(frame => {
+      return {
```

Comment:
Nit: you could avoid the block and the `return` statement by using parens: `frame => ({ vertices: …, })`.